lib/operations/db_collection_ops.js

```diff
@@ -0,0 +1,125 @@
+'use strict';
+
+const applyWriteConcern = require('../utils').applyWriteConcern;
+const Collection = require('../collection');
+const executeCommand = require('./db_execute_commands').executeCommand;
+const f = require('util').format;
+const handleCallback = require('../utils').handleCallback;
+const MongoError = require('mongodb-core').MongoError;
+const ReadPreference = require('mongodb-core').ReadPreference;
+const shallowClone = require('../utils').shallowClone;
+
+// Filter out any write concern options
+const illegalCommandFields = [
+  'w',
+  'wtimeout',
+  'j',
+  'fsync',
+  'autoIndexId',
+  'strict',
+  'serializeFunctions',
+  'pkFactory',
+  'raw',
+  'readPreference',
+  'session'
+];
+
+function createCollection(self, name, options, callback) {
```

Comment:
please document all operations thoroughly. Also consider dropping `self` in these contexts, since it isn't immediately obvious what `self` is in this case (or meant to be)